src/components/mc/cpu/mc_cpu.c

```diff
@@ -9,35 +9,153 @@
 #include "utils/ucc_malloc.h"
 #include <sys/types.h>
 
+static ucc_status_t ucc_mc_cpu_mem_alloc_pool_with_init(ucc_mc_buffer_header_t **ptr,
+                                                   size_t size);
+
 static ucc_config_field_t ucc_mc_cpu_config_table[] = {
     {"", "", NULL, ucc_offsetof(ucc_mc_cpu_config_t, super),
      UCC_CONFIG_TYPE_TABLE(ucc_mc_config_table)},
 
-    {NULL}};
+    {"ELEM_SIZE", "1048576", "The size of each element in mc cpu mpool",
+     ucc_offsetof(ucc_mc_cpu_config_t, cpu_elem_size),
+     UCC_CONFIG_TYPE_MEMUNITS},
+
+    {"MAX_ELEMS", "8", "The max amount of elements in mc cpu mpool",
+     ucc_offsetof(ucc_mc_cpu_config_t, cpu_max_elems), UCC_CONFIG_TYPE_UINT},
 
-static ucc_status_t ucc_mc_cpu_init()
+    {NULL}
+
+};
+
+static ucc_status_t ucc_mc_cpu_init(ucc_thread_mode_t thread_mode)
 {
     ucc_strncpy_safe(ucc_mc_cpu.super.config->log_component.name,
                      ucc_mc_cpu.super.super.name,
                      sizeof(ucc_mc_cpu.super.config->log_component.name));
+    // spinlock assures 1 thread only initiates mpool and allocates its buffers
```

Comment:
general note: if you want to put a comment in the code, then don't add "obvious" comments. For example, here it is obvious that spinlock assures 1 thread accesses the critical section - that is the property of spinlock. However, you could state that it is possible for multiple threads to concurrently execute different collective operations and thus concurrently enter this initialization function - this is why we need a lock here.